webrtc/RTCPeerConnection-helper.js

```diff
@@ -182,16 +182,27 @@ function exchangeIceCandidates(pc1, pc2) {
 }
 
 // Helper function for doing one round of offer/answer exchange
-// betweeen two local peer connections
-function doSignalingHandshake(localPc, remotePc) {
-  return localPc.createOffer()
-  .then(offer => Promise.all([
-    localPc.setLocalDescription(offer),
-    remotePc.setRemoteDescription(offer)]))
-  .then(() => remotePc.createAnswer())
-  .then(answer => Promise.all([
-    remotePc.setLocalDescription(answer),
-    localPc.setRemoteDescription(answer)]))
+// between two local peer connections
+async function doSignalingHandshake(localPc, remotePc, options={}) {
```

Comment:
Ok, but I dislike helper options, as they turn into replacement APIs. Helpers should stay in their lane IMHO.

This one seems ripe for typos:
```js
doSignalingHandshake(pc1, pc2, {modfiyOffer: offer => myRegExp(offer)});
```
The above produces a false positive due to my typo (did you spot it?) - Compare w/spec where typo is impossible:
```js
    await pc1.setLocalDescription(myRegExp(await pc1.createOffer()));
    await pc2.setRemoteDescription(pc1.localDescription);
    await pc2.setLocalDescription(await pc2.createAnswer());
    await pc1.setRemoteDescription(pc2.localDescription);
```
I'd argue the latter API is right for someone needing to mange either the offer or the answer.

As to who needs the former, how much of this is driven by Chrome having  `onnegotiationneeded` bugs?